connectors/src/lib/models/intercom.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  CreationOptional,
+  DataTypes,
+  ForeignKey,
+  InferAttributes,
+  InferCreationAttributes,
+  Model,
+} from "sequelize";
+
+import { Connector, sequelize_conn } from "@connectors/lib/models";
+
+export class IntercomConnectorState extends Model<
+  InferAttributes<IntercomConnectorState>,
+  InferCreationAttributes<IntercomConnectorState>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare lastGarbageCollectionFinishTime?: Date;
+
+  declare connectorId: ForeignKey<Connector["id"]>;
+}
+
+IntercomConnectorState.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    lastGarbageCollectionFinishTime: {
+      type: DataTypes.DATE,
+      allowNull: true,
+    },
+  },
+  {
+    sequelize: sequelize_conn,
+    modelName: "intercom_connector_states",
+    indexes: [{ fields: ["connectorId"], unique: true }],
+  }
+);
+
+Connector.hasOne(IntercomConnectorState);
+
+export class IntercomCollection extends Model<
+  InferAttributes<IntercomCollection>,
+  InferCreationAttributes<IntercomCollection>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare workspaceId: string;
+  declare collectionId: string;
+
+  declare helpCenterId: string | null;
+  declare parentId: string | null;
+
+  declare name: string;
+  declare description: string | null;
+
+  declare connectorId: ForeignKey<Connector["id"]>;
+}
+
+IntercomCollection.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    workspaceId: {
```

Comment:
This is an intercom concept here?

Reply:
Yes indeed it's the intercom workspace id and I agree it is confusing, I will rename!